src/App.tsx

```diff
@@ -14,210 +28,86 @@ export const App: React.FC = () => {
     }
   }, []);
 
+  useEffect(() => {
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos);
+    }
+  }, []);
+
+  const handleEnterPress = (event: KeyboardEvent) => {
+    if (event.key !== 'Enter' || !user) {
+      return;
+    }
+
+    const todo = {
+      title,
+      userId: user.id,
+      completed: false,
+    };
+
+    addTodo(todo)
+      .then(result => setTodos(
+        prevTodos => (prevTodos
+          ? [...prevTodos, result]
+          : [result]),
+      ))
+      .catch(() => {
+        setOnError('add');
+        setIsHidden(false);
+      })
+      .finally(() => {
+        setTitle('');
+        setTimeout(() => setIsHidden(true), 3000);
+      });
+  };
+
+  const handleFilter = (arr: Todo[], filterType: string) => {
+    switch (filterType) {
+      case Filter.active:
+        return arr.filter(element => !element.completed);
+
+      case Filter.completed:
+        return arr.filter(element => element.completed);
+
+      default:
+        return arr;
+    }
+  };
+
+  const visibleTodos = !todos ? [] : handleFilter(todos, filter);
+  const activeCount = !todos ? 0 : todos.filter(todo => !todo.completed).length;
```

Comment:
In which case todo can be falthy? If initial value is empty array which is truthy value